File: app/src/main/java/com/example/fizyoterapi/DataBaseHelper.kt
package com.example.fizyoterapi

import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper



private var Database="DATA_FİZYO"
private var TABLE_HASTA="HASTALAR"
private var TABLE_DOKTOR="DOKTORLAR"
var column_phone="phone"
var cloumn_password="password"

var cloumn_sure="ekran_süresi"

var cloumn_id="id"

class DataBaseHelper (context: Context): SQLiteOpenHelper(context,Database,null,1){
    override fun onCreate(db: SQLiteDatabase?) {
        val createtable = "CREATE TABLE $TABLE_HASTA($cloumn_id integer primary key autoincrement,$column_phone varchar(50) , $cloumn_sure varchar(50),$cloumn_password varchar(50))"
        db?.execSQL(createtable)
        val createtable2 = "CREATE TABLE $TABLE_DOKTOR($column_phone varchar(50) , $cloumn_password varchar(50))"
        db?.execSQL(createtable2)
    }

    override fun onUpgrade(
        db: SQLiteDatabase?,
        oldVersion: Int,
        newVersion: Int
    ) {
        TODO("Not yet implemented")
    }

   fun insertHasta(phone:String,password:String){
       val db = this.writableDatabase
       val cv = ContentValues()
       cv.put(column_phone,phone)
       cv.put(cloumn_password,password)
       db.insert(TABLE_HASTA,null,cv)

   }

  fun insertDoktor(phone:String,password:String){
       val db = this.writableDatabase
       val cv = ContentValues()
       cv.put(column_phone,phone)
       cv.put(cloumn_password,password)
       db.insert(TABLE_DOKTOR,null,cv)

   }


  fun checkHasta(phone:String,password:String):Boolean {
      val db = this.readableDatabase
      val query = "select * from $TABLE_HASTA where $column_phone='$phone' and $cloumn_password='$password'"
      val cursor = db.rawQuery(query, null)
      val exists = cursor.count > 0
      cursor.close()
      return exists
  }

    fun checkDoktor(phone:String,password:String):Boolean {
    val db = this.readableDatabase
    val query = "select * from $TABLE_DOKTOR where $column_phone='$phone' and $cloumn_password='$password'"
    val cursor = db.rawQuery(query, null)
    val exists = cursor.count > 0
    cursor.close()
    return exists

}
    fun getAllHastaPhones(): List<String> {
        val phoneList = mutableListOf<String>()
        val db = this.readableDatabase
        val cursor: Cursor = db.rawQuery("SELECT $column_phone FROM $TABLE_HASTA", null)

        if (cursor.moveToFirst()) {
            do {
                val phone = cursor.getString(cursor.getColumnIndexOrThrow(column_phone))
                phoneList.add(phone)
            } while (cursor.moveToNext())
        }

        cursor.close()
        db.close()
        return phoneList
    }





}

File: lib/services/database_helper.dart
// lib/services/database_helper.dart (DÜZELTİLMİŞ VE KESİN ÇÖZÜM)

import 'package:sqflite/sqflite.dart';
import 'package:path/path.dart';

class DatabaseHelper {
  static const _databaseName = "DATA_FİZYO";
  static const _databaseVersion = 1;

  static const tableHasta = 'HASTALAR';
  static const tableDoktor = 'DOKTORLAR';

  static const columnPhone = 'phone';
  static const columnPassword = 'password';
  static const columnSure = 'ekran_süresi';
  static const columnId = 'id';

  DatabaseHelper._privateConstructor();
  static final DatabaseHelper instance = DatabaseHelper._privateConstructor();

  static Database? _database;
  Future<Database> get database async {
    if (_database != null) return _database!;
    _database = await _initDatabase();
    return _database!;
  }

  _initDatabase() async {
    String path = join(await getDatabasesPath(), _databaseName);
    return await openDatabase(path,
        version: _databaseVersion,
        onCreate: _onCreate);
  }

  Future _onCreate(Database db, int version) async {
    await db.execute('''
          CREATE TABLE $tableHasta (
            $columnId INTEGER PRIMARY KEY AUTOINCREMENT,
            $columnPhone TEXT NOT NULL,
            $columnSure TEXT,
            $columnPassword TEXT NOT NULL
          )
          ''');
    await db.execute('''
          CREATE TABLE $tableDoktor (
            $columnPhone TEXT NOT NULL,
            $columnPassword TEXT NOT NULL
          )
          ''');
  }

  Future<int> insertHasta(String phone, String password) async {
    final db = await instance.database; // db tanımlı
    return await db.insert(tableHasta, {columnPhone: phone, columnPassword: password});
  }

  Future<bool> checkHasta(String phone, String password) async {
    final db = await instance.database; // db tanımlı
    List<Map> result = await db.query(
      tableHasta,
      columns: [columnPhone],
      where: '$columnPhone = ? AND $columnPassword = ?',
      whereArgs: [phone, password],
    );
    return result.isNotEmpty;
  }
  
  Future<int> insertDoktor(String phone, String password) async {
    final db = await instance.database; // db tanımlı
    return await db.insert(tableDoktor, {columnPhone: phone, columnPassword: password});
  }

  Future<bool> checkDoktor(String phone, String password) async {
    final db = await instance.database; // db tanımlı
    List<Map> result = await db.query(
      tableDoktor,
      columns: [columnPhone],
      where: '$columnPhone = ? AND $columnPassword = ?',
      whereArgs: [phone, password],
    );
    return result.isNotEmpty;
  }
  
  // ⭐️ HATA DÜZELTİLDİ: 'db' değişkeni tanımlandı
  Future<List<String>> getAllHastaPhones() async {
    final db = await instance.database; // Eksik olan satır eklendi.
    final List<Map<String, dynamic>> maps = await db.query(tableHasta, columns: [columnPhone]);
    
    return List.generate(maps.length, (i) {
      return maps[i][columnPhone] as String;
    });
  }
}